src/BaseIO.hpp

```diff
@@ -292,6 +292,13 @@ class BaseIO
   virtual std::unique_ptr<BaseRecordingData> getDataSet(
       const std::string& path) = 0;
 
+  /**
+   * @brief Checks whether an object already exists at the location in the file.
```

Comment:
```suggestion
   * @brief Checks whether a Dataset, Group, or Link already exists at the location in the file.
```

``H5Lexists`` only works for Datasets and Groups (but not Attributes). I don't think this function needs to cover attributes right now, but I figured it would be useful to clarify this in the docstring.